src/collectors/jolokia/kafka_jolokia.py

```diff
@@ -0,0 +1,592 @@
+"""
+Collects Kafka JMX metrics from the Jolokia Agent.  Extends the
+JolokiaCollector to reformat the Kafka MBean names into easier to
+understand metric names.
+
+Additionally, we also connect to Zookeeper to collect certain 
+metrics that are critical to Kafka monitoring.  This is done in
+two ways:
+    1) By extending the ProcessCollector, we can execute a command-
+        line function against the Zookeeper instance to extract 
+        infomation about the consumer groups and their corresponding
+        consumer lag.  This replaces and improves upon the 
+        KafkaConsumerLagCollector.
+    2) By extending the ZookeeperCollector, we can collect the 
+        key Zookeeper metrics for Kafka without requiring the user
+        to configure a separate collector. This does not replace
+        the Zookeeper collector, however; users with a Zookeeper
+        instance being used for something other than Kafka can still
+        use the Zookeeper collector on its own. 
+
+Netuitive Change History
+    2016/10/26 DVG - Initial version.
+    2016/11/16 DVG - Added debug logging, additional error handling, refactoring of common code across the Kafka 8 and 9 handling.
+                     Fixed bug with Zookeeper collector defaulting to localhost:2181
+"""
+
+from jolokia import JolokiaCollector
+from zookeeper import ZookeeperCollector
+from diamond.collector import ProcessCollector
+import math
+import string
+import re
+
+class KafkaJolokiaCollector(JolokiaCollector, ProcessCollector, ZookeeperCollector):
+
+    ########
+    #
+    # The get_default_config_help() function is over-ridden to provide help on the config
+    # options specific to this colector.
+    #
+    # 2016-11-03 DVG
+    #
+    ########
+
+    def get_default_config_help(self):
+
+        # Get the parent class config help object.
+        config_help = super(KafkaJolokiaCollector, self).get_default_config_help()
+         
+        # Update it....
+        config_help.update({
+            'bin': 'The path to the kafka-run-class.sh script.  Default is /opt/kafka/bin/kafka-run-class.sh',
+            'version': 'The version of Kafka. Two values are accepted here. 8 for Kafka versions 8 and earlier, or 9 for Kafka versions 9 or later. The default is 9.',
```

Comment:
I did look into auto-detecting the version and there doesn't seem to be a very clean way of doing it.  The "0." seemed redundant to me, but you're right, we should stick to the official numbering. I'll update the code to: a) require this variable to be set (i.e., no default value); b) accept values of 0.8, 0.9, and 0.10 - and throw an error on anything else.